guide/keyv/README.md

```diff
@@ -0,0 +1,168 @@
+
+```bash
+npm install --save @keyv/redis
+npm install --save @keyv/mongo
+npm install --save @keyv/sqlite
+npm install --save @keyv/postgres
+npm install --save @keyv/mysql
+```
+
+Now that Keyv and any necessary drivers are installed, create an instance of Keyv.
+
+<!-- eslint-skip -->
+```js
+const Keyv = require('keyv');
+
+// One of the following
+const keyv = new Keyv(); // for in-memory storage
+const keyv = new Keyv('redis://user:pass@localhost:6379');
+const keyv = new Keyv('mongodb://user:pass@localhost:27017/dbname');
+const keyv = new Keyv('sqlite://path/to/database.sqlite');
+const keyv = new Keyv('postgresql://user:pass@localhost:5432/dbname');
+const keyv = new Keyv('mysql://user:pass@localhost:3306/dbname');
+```
+
+Make sure to handle connection errors.
+
+```js
+keyv.on('error', err => console.error('Keyv connection error:', err));
+```
+
+For more detailed setup, check out the [Keyv readme](https://github.com/lukechilds/keyv/blob/master/README.md).
+
+## Usage
+
+Keyv exposes a familiar [Map](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map)-like API. However, it only has `set`, `get`, `delete`, and `clear` methods. Additionally, instead of immediately returning data, these methods return [Promises](/additional-info/async-await) that resolve with the data.
+
+```js
+(async () => {
+	// true
+	await keyv.set('foo', 'bar');
+
+	// bar
+	await keyv.get('foo');
+
+	// undefined
+	await keyv.clear();
+
+	// undefined
+	await keyv.get('foo');
+})();
+```
+
+## Application
+
+Although Keyv can be used in any scenario where you need key-value data, we will focus on setting up a per-guild prefix configuration using Sqlite.
+
+<tip>This section will still work with any provider supported by Keyv. We recommend PostgreSQL for larger applications.</tip>
+
+### Setup
+
+```js
+const Discord = require('discord.js');
+const Keyv = require('keyv');
+
+const client = new Discord.Client();
+const prefixes = new Keyv('sqlite://path/to.sqlite');
+const globalPrefix = '.';
+```
+
+### Command handler
+
+This guide uses a very basic command handler with some added complexity to allow for multiple prefixes. For more a more robust command handler, look at the [command handling](/command-handling) guide.
+
+```js
+client.on('message', async (message) => {
+	if (message.author.bot) return;
+
+	let args;
+	// handle messages in a guild
+	if (message.guild) {
+		let prefix;
+
+		// check if the message starts with the global prefix
+		if (message.content.startsWith(globalPrefix)) {
+			prefix = globalPrefix;
+		}
+		else {
+			// check the guild-level prefix
+			const guildPrefix = await prefixes.get(message.guild.id);
+			if (message.content.startsWith(guildPrefix)) prefix = guildPrefix;
+		}
+
+		// if we found a prefix, setup args; otherwise, this isn't a command
+		if (prefix) args = message.content.slice(prefix.length).split(/\s+/);
+		else return;
```

Comment:
Not a fan of `else return` and long one-line if statements. I'd be more consistent if it were written as
```js
if (!prefix) return;
args = message.content.slice(prefix.length).split(/\s+/);